common/components/org.wso2.carbon.analytics.util/src/main/java/org/wso2/carbon/analytics/common/spark/udf/RegExrUDF.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.analytics.common.spark.udf;
+
+import org.wso2.carbon.analytics.spark.core.udf.CarbonUDF;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Functions to extract message filtered by regex pattern
+ */
+public class RegExrUDF implements CarbonUDF {
+    public String getRegexMatch(String message, String regEx) {
+        Pattern pattern = Pattern.compile(regEx);
```

Comment:
Hi Sajith,
We need to check null on message, regEx. It throws NPE otherwise.

Its not the return null I was worried. The fix to return "" is artificial. return null is OK IMO.
So I am -1 on the fix.

Cheers,
Ruwan